internal/codegen/templates/delegate.tmpl

```diff
@@ -158,3 +171,48 @@ func (m *{{.Name | toLower}}CallbacksMap) delete(p unsafe.Pointer) {
 
 	delete(m.callbacks, p)
 }
+
+// typedEventHandlerReleaseChannelsMap keeps a map with channels
+// used to keep a goroutine alive during the lifecycle of this object.
+// This is required to avoid causing a deadlock error.
+// See this: https://github.com/golang/go/issues/55015
+type {{.Name | toLower}}ReleaseChannelsMap struct {
+	mu        *sync.Mutex
+	chans map[unsafe.Pointer]chan struct{}
+}
+
+func (m *{{.Name | toLower}}ReleaseChannelsMap) create(p unsafe.Pointer) {
```

Comment:
Small nit, `Map` seems like an unneeded suffix.

Reply:
Also, does it make sense to define this as "acquire" more than "create"? This way it seems to match better the `acquire`+`release`.

Reply:
Agree on both 😄